components/TransactionList/index.js

```diff
@@ -1,13 +1,73 @@
+import { Fragment } from "react";
 import TransactionCard from "../TransactionCard";
-import { Ul } from "./TransactionList.styled";
+import * as Styled from "./TransactionList.styled";
+import {
+  findYearsInTransactions,
+  filterTransactionsByYear,
+  filterTransactionsByMonth,
+  sortMonths,
+} from "../../utils/transaction";
+import { months } from "@/utils/months";
 
 function TransactionList({ transactions }) {
+  const yearsInTransactions = findYearsInTransactions(transactions);
+
+  const transactionsPerYearsAndMonths = yearsInTransactions.map(
+    (year, index) => {
+      const transactionsPerYear = filterTransactionsByYear(transactions, year);
+      if (transactionsPerYear.length === 0) {
+        return null;
```

Comment:
you don't check if the year element is null or not in your jsx mapping later on. This will lead to an error if the value is actually null, since you try to access the `transactionsPerMonth` which does not exist if the element is null. If you want to remove the month entirely, you need to use filter to remove the elements which are "null".

Reply:
As far as I see the code passage it not necessary at all, as beforehands only years that are in the data are filtered. So I removed it.